acsf_tools.drush.inc

```diff
@@ -213,6 +227,57 @@ function drush_acsf_tools_dump() {
   }
 }
 
+/**
+ * Action callback for acsf-tools-restore command.
+ */
+function drush_acsf_tools_restore() {
+  // Ask for confirmation before running the command.
+  if (!_drush_acsf_tools_prompt_confirm()) {
+    return;
+  }
+
+  // Identify source folder.
+  $source_folder = drush_get_option('source-folder');
+  if (!isset($source_folder)) {
+    $source_folder = '~/drush-backups';
+  }
+
+  if (!is_dir($source_folder)) {
+    // Source folder does not exist..
+    drush_log("\nSource folder $source_folder does not exist.", 'error');
+    return;
+  }
+
+  // Look for list of sites and loop over it.
+  if ($sites = _drush_acsf_tools_get_sites()) {
+    $arguments = drush_get_arguments();
+
+    $options = drush_get_context('cli');
+    unset($options['php']);
+    unset($options['php-options']);
+    unset($options['source-folder']);
+
+    foreach ($sites as $details) {
+      $domain = $details['domains'][0];
+      $prefix = explode('.', $domain)[0];
+
+      $source_file = $source_folder . '/' . $prefix . '.sql';
+
+      if (!file_exists($source_file)) {
+        drush_log("\n => No source file $source_file for $prefix site.", "error");
+        continue;
+      }
+
+      $command_drop = 'sql-drop';
+      $command_cli = 'sql-cli < ' . $source_file;
+
+      drush_log("\n=> Dropping and restoring database on $domain", 'ok');
+      drush_invoke_process('@self', $command_drop, $arguments, $options + ['l' => $domain, 'y']);
+      drush_invoke_process('@self', $command_cli, $arguments, $options + ['l' => $domain]);
+    }
+  }
+}
```

Comment:
There may be a need of flushing memcache when at least one site got restored

Reply:
@sdelbosc I am wondering if flushing memcache should not be the role of a different command. I would prefer to not block this command given there may be discussion around the need and best way to flush memcache.

Reply:
fine for me